yash-datalake/azuredeploy.json

```diff
@@ -0,0 +1,576 @@
+      },
+      "defaultValue": 1
+    },
+    "isAutoInflateEnabled": {
+      "type": "string",
+      "allowedValues": [
+        "True",
+        "False"
+      ],
+      "metadata": {
+        "description": "Enable or disable AutoInflate"
+      },
+      "defaultValue": "True"
+    },
+    "messageRetentionInDays": {
+      "type": "int",
+      "minValue": 1,
+      "maxValue": 7,
+      "metadata": {
+        "description": "How long to retain the data in Event Hub"
+      },
+      "defaultValue": 1
+    },
+    "partitionCount": {
+      "type": "int",
+      "minValue": 2,
+      "maxValue": 32,
+      "metadata": {
+        "description": "Number of partitions chosen"
+      },
+      "defaultValue": 2
+    },
+    "captureEncodingFormat": {
+      "type": "string",
+      "allowedValues": [
+        "Avro"
+      ],
+      "metadata": {
+        "description": "The encoding format Eventhub capture serializes the EventData when archiving to your storage"
+      },
+      "defaultValue": "Avro"
+    },
+    "adlAnalyticsName": {
+      "type": "string",
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datalakedemoadlaa'))]",
+      "metadata": {
+        "description": "The name of the Data Lake Analytics account to create."
+      }
+    },
+    "adlStoreName": {
+      "type": "string",
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datalakedemoadls1a'))]",
+      "metadata": {
+        "description": "The name of the Data Lake Store account to create."
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "Size of vm Eg. Standard_D1_v2"
+      },
+      "defaultValue": "Standard_D1_v2"
+    },
+    "vm_username": {
+      "type": "string",
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'azuredemo'))]",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "vm_password": {
+      "type": "string",
+      "defaultValue": "Yash@1234",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "OptionalWizardInstall": {
+      "type": "string",
+      "allowedValues": [
+        "Yes",
+        "No"
+      ],
+      "defaultValue": "Yes",
+      "metadata": {
+        "description": "Select whether the VM should be in production or not."
+      }
+    },
+    "dataFactoryName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the data factory. Must be globally unique."
+      },
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datafactorya'))]"
+    },
+    "servicePrincipalId": {
+      "type": "string",
+      "defaultValue": "null",
+      "metadata": {
+        "description": "The ID of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "servicePrincipalKey": {
+      "type": "string",
+      "defaultValue": "null",

```

Comment:
use a emptystring or remove the defaultValue

Reply:
Changed defaultValue to emptystring